pycloudlib/ibm/cloud.py

```diff
@@ -261,6 +261,42 @@ def get_or_create_vpc(self, name: str) -> VPC:
             self.created_vpcs.append(vpc)
             return vpc
 
+    def _choose_from_existing_floating_ips(
+        self, name_includes="default-floating-ip"
+    ) -> dict:
+        """
+        Choose a floating IP from existing floating IPs
+
+        Args:
+            name_includes: string, name to filter floating IPs by
+
+        Returns:
+            A floating IP object
+        """
+        floating_ips = list(
+            _iter_resources(
+                self._client.list_floating_ips,
+                resource_name="floating_ips",
+                # Select that match the desired name
+                filter_fn=lambda ip: name_includes in ip["name"],
+            )
+        )
+        if not floating_ips:
+            raise IBMException("No floating IPs found matching substring %s", name_includes)
```

Comment:
This syntax is for loggers. One way to correctly interpolate strings would be:

```suggestion
            raise IBMException(f"No floating IPs found matching substring `{name_includes}`")
```